### DISCUSSION TITLE
REST API CourtListener OCR text access

### DISCUSSION THREAD

**Discussion by kingxia** [freelawproject/courtlistener#2650] [Q&A] [answered]
Hi! I've been tinkering with CourtListener's data access features and I was wondering if there is a way to fetch the OCR text that RECAP has for documents.

For example, [Purcell v. Spokeo, Inc. document #12](https://www.courtlistener.com/docket/4177509/12/purcell-v-spokeo-inc/) has a PDF view and a text view. I see that if I request `/api/rest/v3/dockets/4177509/`, the `filepath_ia_json` field points me to [a list of available documents](https://ia800503.us.archive.org/7/items/gov.uscourts.cand.231495/gov.uscourts.cand.231495.docket.json), which includes this #12 document. The #12 document includes an internet archive link (`filepath_ia`) to the original PDF, but I don't see a way to access the derived text.

It's entirely possible that I've overlooked how to get this text, or that it lives in another one of the data sources (e.g. replication or bulk data) that CourtListener provides. It's also possible I misunderstand the relationship between CourtListener and RECAP! While running OCR over the provided documents on my end is entirely reasonable, I'm wondering if there is a way to not reinvent the wheel here.

Thanks so much!

**Comment by mlissner** [comment, marked as answer]
Yes, this text is available, as well as a number of other fields, via the `recap-documents` endpoint. I can't tell from your Github username if we've given you access to that, but it's usually something we provide after we learn a bit more about the way people intend to use the data. 

Can you drop us an email and we'll get the ball rolling that way?

**Reply by kingxia** [reply]
Thanks for the quick reply - and for the automated hello email! I don't have anything beyond the basic public access right now; it makes sense you have graduated access. I'll shoot you an email (I take it the [Contact page](https://www.courtlistener.com/contact/) is the preferred method here) and we'll go from there!
